### PR TITLE
Allow user to use local scripts and configuration files in lieu of generated when using openshift ocp.sh script.

### PR DESCRIPTION
**Pull request by JamesDrummond**
### What does this PR do?
Allow user to local scripts and configuration files in lieu of generated by setting CHE_OPENSHIFT_GENERATE_SCRIPTS =false. Default will remain to generate scripts DEFAULT_CHE_OPENSHIFT_GENERATE_SCRIPTS =true.

### What issues does this PR fix or reference?
NA

#### Release Notes
Allow user to use local scripts and configuration files in lieu of generated when using openshift ocp.sh script.

Signed-off-by: James Drummond james@devcomb.com

### CODE REVIEW

File: dockerfiles/init/modules/openshift/files/scripts/ocp.sh
Comment (riuvshin) [inline dockerfiles/init/modules/openshift/files/scripts/ocp.sh]: that is copy paste based on older version of script, please update it accordingly to the current state.
And personally, I don't like the idea of allowing to use local scripts which is error prone.

Comment (JamesDrummond) [reply dockerfiles/init/modules/openshift/files/scripts/ocp.sh]: I don't like the idea for people not developing che using but for me it is much easier to use local scripts versus generated. 

Will make the change of the copy paste issue.

Comment (riuvshin) [reply dockerfiles/init/modules/openshift/files/scripts/ocp.sh]: there is a reason why we generate this, we can merge this but there should be some comments that this is errorprone and may not work as expected

Comment (JamesDrummond) [reply dockerfiles/init/modules/openshift/files/scripts/ocp.sh]: Added warning comments.

**Review by riuvshin — APPROVED** [review]
before merge please, test carefully single / multi user mode and deploying from scratch

**Comment by JamesDrummond** [comment]
@benoitf I need another approval before I can merge. Let me know if 2 approvals are still needed for each PR per eclipse policy.

**Comment by benoitf** [comment]
@JamesDrummond well the PR has conflicting files

**Comment by JamesDrummond** [comment]
@benoitf Reply beat my fix :)

**Comment by JamesDrummond** [comment]
@benoitf @sleshchenko Can one of you review/approve?

**Comment by sleshchenko** [comment]
I tried to use your changes from your branch, so I execute
```
export CHE_GENERATE_SCRIPTS="false"
./deploy_che.sh --deploy-che
```
And I got the following output. As you can see `deploy_che.sh` script is invoked twice. Do you get the same or I did something wrong?

<details>
<summary>OCP output</summary>
<pre>
[serg@sleschenko scripts]$ ./ocp.sh --deploy-che
OCP using existing scripts in current folder.

                10100000                
            0110011001101101            
       01001001001001011011010110       
   1100110101010001000001011101000001   
0111001001111010111011111000111100010001
1110111011010111000001100000111001110   
101010011001001001    10011101000       
1011000101000              0            
100101011                           0110
10100001                        11011111
111101111011                111011110001
10111001000110101      00000111111011111
0110111111011100000001011001010101111111
1101000001101001000010010111101011011011
    00000001111001000001110111100000    
        000110101101000100011110        
            1101011100001001            
                 000101                 


[CHE] Logging on using OpenShift endpoint "https://172.19.20.21:8443"...done!
[CHE] Checking if project "eclipse-che" exists...Project does not exist...creating it...done!
[CHE] Switching to "eclipse-che"...done!
[CHE] Setting admin role to "che" service account...rolebinding "che" created
[CHE] Retrieving latest version of fabric8 tenant Che template...done! (v.2.0.78)

[CHE] Deploying Che on ocp (image eclipse/che-server:nightly)
serviceaccount "che" created
service "che-host" created
persistentvolumeclaim "che-data-volume" created
persistentvolumeclaim "claim-che-workspace" created
deploymentconfig "che" created
route "che" created


[CHE] Che deployment has been successufully bootstrapped
[CHE] -> To check OpenShift deployment logs: 'oc get events -w'
[CHE] -> To check Che server logs: 'oc logs -f dc/che'
[CHE] -> Once the deployment is completed Che will be available at: 
[CHE]    http://che-eclipse-che.172.19.20.21.nip.io


rm: can't remove '/to_remove': Resource busy
WARN: Did detect TTY but not in interactive mode
INFO: (che cli): nightly - using docker 17.12.0-ce / native
WARN: (che init): 'nightly' installations cannot be upgraded to non-nightly versions
INFO: (che init): Installing configuration and bootstrap variables:
INFO: (che init):   CHE_HOST=172.19.20.21
INFO: (che init):   CHE_VERSION=nightly
INFO: (che init):   CHE_CONFIG=/tmp/che-config
INFO: (che init):   CHE_INSTANCE=/tmp/che-config/instance
INFO: (che config): Generating che configuration...
INFO: (che config): Customizing docker-compose for running in a container

                10100000                
            0110011001101101            
       01001001001001011011010110       
   1100110101010001000001011101000001   
0111001001111010111011111000111100010001
1110111011010111000001100000111001110   
101010011001001001    10011101000       
1011000101000              0            
100101011                           0110
10100001                        11011111
111101111011                111011110001                                                                                                                                                                                                     
10111001000110101      00000111111011111                                                                                                                                                                                                     
0110111111011100000001011001010101111111
1101000001101001000010010111101011011011
    00000001111001000001110111100000    
        000110101101000100011110        
            1101011100001001            
                 000101                 


[CHE] Logging on using OpenShift endpoint "https://172.19.20.21:8443"...done!
[CHE] Checking if project "eclipse-che" exists...done!
[CHE] Switching to "eclipse-che"...done!
[CHE] Setting admin role to "che" service account...rolebinding "che" configured
[CHE] Retrieving latest version of fabric8 tenant Che template...done! (v.2.0.78)

[CHE] Deploying Che on ocp (image eclipse/che-server:nightly)
serviceaccount "che" configured
service "che-host" configured
persistentvolumeclaim "che-data-volume" configured
persistentvolumeclaim "claim-che-workspace" configured
deploymentconfig "che" configured
route "che" configured


[CHE] Che deployment has been successufully bootstrapped
[CHE] -> To check OpenShift deployment logs: 'oc get events -w'
[CHE] -> To check Che server logs: 'oc logs -f dc/che'
[CHE] -> Once the deployment is completed Che will be available at: 
[CHE]    http://che-eclipse-che.172.19.20.21.nip.io


[CHE] wait CHE pod booting...
[serg@sleschenko scripts]$ ^C
</pre>
</details>

P.S. The same with `CHE_GENERATE_SCRIPTS=true`.

**Comment by JamesDrummond** [comment]
@sleshchenko Sorry it's my fault after I merged master changes from master didn't catch I had deploy part in there twice and out of generate script if statement. Fixed now.

File: dockerfiles/init/modules/openshift/files/scripts/ocp.sh
Comment (sleshchenko) [inline dockerfiles/init/modules/openshift/files/scripts/ocp.sh]: `${CONFIG_DIR}/instance/config/openshift/scripts/` is used twice. Please consider moving it into a separated variable like `OPENSHIFT_SCRIPTS_FOLDER`

File: dockerfiles/init/modules/openshift/files/scripts/ocp.sh
Comment (sleshchenko) [inline dockerfiles/init/modules/openshift/files/scripts/ocp.sh]: The previous message contains information about `configuration files` but this one doesn't.  Should we add it here too?
To be sure that I understand `configuration files` correctly: is it about yaml files?

Comment (riuvshin) [reply dockerfiles/init/modules/openshift/files/scripts/ocp.sh]: it is shell scripts and yaml files needed to deploy che multi user

File: dockerfiles/init/modules/openshift/files/scripts/ocp.sh
Comment (riuvshin) [inline dockerfiles/init/modules/openshift/files/scripts/ocp.sh]:
```
 +      if [ $IMAGE_PULL_POLICY == "Always" ]; then
 +         docker pull "$IMAGE_INIT"
 +      fi
```
this must be moved out from `if` statement because it should be considered in both cases

**Review by riuvshin — CHANGES_REQUESTED** [review]

File: dockerfiles/init/modules/openshift/files/scripts/ocp.sh
Comment (JamesDrummond) [reply dockerfiles/init/modules/openshift/files/scripts/ocp.sh]: Moved.

File: dockerfiles/init/modules/openshift/files/scripts/ocp.sh
Comment (JamesDrummond) [reply dockerfiles/init/modules/openshift/files/scripts/ocp.sh]: Added `configuration files` to second mesage.

File: dockerfiles/init/modules/openshift/files/scripts/ocp.sh
Comment (JamesDrummond) [reply dockerfiles/init/modules/openshift/files/scripts/ocp.sh]: Added variable OPENSHIFT_SCRIPTS_FOLDER.

**Comment by JamesDrummond** [comment]
@riuvshin @sleshchenko Updates complete. Please review.

**Review by sleshchenko — APPROVED** [review]

**Review by riuvshin — APPROVED** [review]

**Pull request merged**
